src/main/java/com/worksap/nlp/lucene/sudachi/ja/SudachiTokenizer.kt

```diff
@@ -42,31 +44,57 @@ class SudachiTokenizer(
   }
 
   private var iterator: MorphemeIterator = MorphemeIterator.EMPTY
+  private var offset = 0
+  private var endOffset = 0
 
   override fun reset() {
     super.reset()
-    var iter = tokenizer.tokenize(input)
-    if (discardPunctuation) {
-      iter = NonPunctuationMorphemes(iter)
-    }
-    iterator = iter
+    iterator = MorphemeIterator.EMPTY
+    offset = 0 // pos from the beginning to current chunk.
+    endOffset = 0
   }
 
   override fun incrementToken(): Boolean {
     clearAttributes()
-    val m = iterator.next() ?: return false
+
+    var m = iterator.next()
+    if (m == null) {
+      // To cope with huge text, it split into chunks (1 MB) for tokenize.
+      // TODO: Should split with meaningful delimitations instead of fixed size (1 MB).
+      val buffer = CharBuffer.allocate(1024)
+      // val buffer = CharBuffer.allocate(1 * 1024 * 1024)
+      val nread = input.read(buffer)
```

Comment:
Let's use `com.worksap.nlp.sudachi.IOTools` here:
`val nread = IOTools.readAsMuchAsCan(input, buffer)`

Reply:
I applied the changes.